### ISSUE TITLE
Ability to fire the media browser from any card as an action

### ISSUE THREAD

**Issue by RomRider** [home-assistant/frontend#7145]
## The request
I'd like to be able to display the media browser from a `media_player` entity from any card. Right now, firing the `show-dialog` event with the `dialogTag: 'dialog-media-player-browse'` doesn't work because the media browser is not loaded. If it is loaded first though a standard `media_player` card, it works. I wouldn't want to duplicate the media browser in a custom card.

## The alternatives
None

## Additional information

Something like this below currently doesn't work in a custom-card unless it has been displayed before through a core media_player card. `dialogImport` is omitted as it would require to duplicate all the code from the media browser which is far from ideal.
```ts
  private _handleMediaBrowerEvent(actionConfig: MediaBrowserActionConfig): void {
    fireEvent(this, 'show-dialog', {
      dialogTag: 'dialog-media-player-browse',
      dialogParams: {
        action: 'play',
        entityId: actionConfig.entity_id ? actionConfig.entity_id : this._config?.entity,
        mediaPickedCallback: (pickedMedia: any) => {
          this._hass!.callService('media_player', 'play_media', {
            entity_id: actionConfig.entity_id ? actionConfig.entity_id : this._config?.entity,
            media_content_id: pickedMedia.item.media_content_id,
            media_content_type: pickedMedia.item.media_content_type,
          });
        },
      },
    });
  }
```

**Comment by 1ntroduc3** [comment]
We;d love to have this feature. So we can launch the mediabrowser from another side.

**Comment by RomRider** [comment]
An approach would be a service type like `more-info` but for it would be called `media-browser`.
@bramkragten, what do you think?

**Comment by 1ntroduc3** [comment]
Is there any update on this feature? This would add great possibilities.

**Comment by ncodee** [comment]
+1, hopefully this is feasible and gets implemented soon.

**Comment by bramkragten** [comment]
> An approach would be a service type like `more-info` but for it would be called `media-browser`.
> @bramkragten, what do you think?

That would not be the way to go, if we wanted to implement this, we should expose the `showMediaBrowserDialog` in `custom-card-helpers`. But this is always a tricky point, as it will hinder our development since we now have to consider custom cards when making changes.

https://github.com/home-assistant/frontend/blob/add-fallback-to-serviceworker/src/components/media-player/show-media-browser-dialog.ts#L15

**Comment by 1ntroduc3** [comment]
Please do consider it. It would make so much great use cases

**Comment by RomRider** [comment]
I'd not consider custom-cards as a blocker for this when making changes, but notifying in some ways that changes are happening would be great (as it is for every other accessible components). Then I'd say it's up to the custom-card's dev to figure out what those changes are.

**Comment by 1ntroduc3** [comment]
> I'd not consider custom-cards as a blocker for this when making changes, but notifying in some ways that changes are happening would be great (as it is for every other accessible components). Then I'd say it's up to the custom-card's dev to figure out what those changes are.
> 
> I'd rather have the option and it breaking in a custom card at some point than not having the option at all.

I agree. This also works for other custom functions that are made available.
